plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlVectorType.java

```diff
@@ -181,13 +184,54 @@ void testVectorUnsupportedDimension()
     void testEuclideanDistanceCompatibility()
     {
         // Verify compatibility between Trino euclidean_distance function and pgvector <-> operator
-        try (TestTable table = new TestTable(postgreSqlServer::execute, "test_vector", "(v vector(3))");
-                TestView view = new TestView(postgreSqlServer::execute, "test_euclidean_distance", "SELECT v <-> '[4,5,6]' FROM " + table.getName())) {
-            postgreSqlServer.execute("INSERT INTO " + table.getName() + " VALUES ('[1,2,3]')");
+        try (TestTable table = new TestTable(postgreSqlServer::execute, "test_vector", "(id int, v vector(3))");
+                TestView view = new TestView(postgreSqlServer::execute, "test_euclidean_distance", "SELECT v <-> '[7,8,9]' FROM " + table.getName())) {
+            postgreSqlServer.execute("INSERT INTO " + table.getName() + " VALUES (1, '[1,2,3]'), (2, '[4,5,6]')");
 
-            assertThat(query("SELECT euclidean_distance(v, ARRAY[4,5,6]) FROM " + table.getName()))
-                    .matches("SELECT * FROM tpch." + view.getName());
+            assertThat(query("SELECT euclidean_distance(v, ARRAY[7,8,9]) FROM " + table.getName()))
+                    .matches("SELECT * FROM tpch." + view.getName())
+                    .isFullyPushedDown();
 
+            assertThat(query("SELECT id FROM " + table.getName() + " ORDER BY euclidean_distance(v, ARRAY[7,8,9]) LIMIT 1"))
+                    .isFullyPushedDown();
+
+            assertThatThrownBy(() -> postgreSqlServer.execute("SELECT v <-> '[NaN]' FROM " + table.getName()))
```

Comment:
why postgresql implementation is tested?

Reply:
I added these assertions to catch future changes of pgvector. I can't find such spec that defines the expected behavior with special values.